packages/maskbook/src/plugins/Trader/UI/trending/SearchResultView.tsx

```diff
@@ -154,14 +156,9 @@ export function SearchResultView(props: SearchResultViewProps) {
     const canSwap = trending.coin.eth_address || trending.coin.symbol.toLowerCase() === 'eth'
     const swapTabIndex = dataProvider !== DataProvider.UNISWAP ? 3 : 1
     const fromToken = chainId === ChainId.Mainnet && coin.is_mirrored ? UST : createEtherToken(chainId)
+    const { decimals } = coinDetailed ?? trending.coin
     const toToken = trending.coin.eth_address
-        ? createERC20Token(
-              chainId,
-              coinDetailed.address,
-              coinDetailed.decimals,
-              coinDetailed.name ?? '',
-              coinDetailed.symbol ?? '',
-          )
+        ? createERC20Token(chainId, coin.eth_address ?? ETH_ADDRESS, decimals ?? 0, coin.name ?? '', coin.symbol ?? '')
```

Comment:
The portion after `??` is unnecessary.

```suggestion
        ? createERC20Token(chainId, coin.eth_address, decimals ?? 0, coin.name ?? '', coin.symbol ?? '')
```

Reply:
ok

Reply:
Cause it's leading by `trending.coin.eth_address`. That is the same as `coin.eth.address`.

Reply:
ok